VideoLocker/src/main/java/org/edx/mobile/util/SecurityUtil.java

```diff
@@ -0,0 +1,50 @@
+package org.edx.mobile.util;
+
+
+import android.content.Context;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+import android.support.annotation.NonNull;
+
+import org.edx.mobile.logger.Logger;
+import org.edx.mobile.module.db.DbStructure;
+import org.edx.mobile.module.prefs.PrefManager;
+
+import java.io.File;
+import java.util.Collections;
+
+public abstract class SecurityUtil {
+    private static final Logger logger = new Logger(SecurityUtil.class);
+
+    public static void clearUserData(@NonNull Context context) {
+        // Clear the App's directory
+        PackageManager packageManager = context.getPackageManager();
+        try {
+            PackageInfo packageInfo = packageManager.getPackageInfo(context.getPackageName(), 0);
+            File appDir = new File(packageInfo.applicationInfo.dataDir);
+            if (appDir.exists() && appDir.isDirectory()) {
+                File[] children = appDir.listFiles();
+                for (int size = children.length, i = 0; i < size; i++) {
+                    FileUtil.deleteRecursive(children[i],
+                            Collections.singletonList(DbStructure.NAME));
```

Comment:
Why not do the `deleteRecursive()` call on the root directory itself, instead of iterating over all it's children? As long as there is an undeleted file in it, the root shouldn't actually be deleted itself.